src/permuter.py

```diff
@@ -261,10 +273,12 @@ def seed_iterator(self) -> Iterator[int]:
             return itertools.repeat(self._force_seed)
         return iter([self._force_seed])
 
-    def try_eval_candidate(self, seed: int) -> EvalResult:
+    def try_eval_candidate(self, seed: int) -> Optional[EvalResult]:
```

Comment:
EvalResult is already an "either a result or an error", wrapping it in an additional layer of "might not exist" feels complex/wrong. Can we either extend EvalResult with an additional enum kind, or save the result in the cache and return that?

Reply:
you're right those options make more sense.
I can do one of those options if you want.  
But how do you feel about me changing it back to EvalResult (Not Optional)
and doing this:
`return EvalError(exc_str="Duplicate Source from Randomization - not scoring", seed=self._cur_seed)`